projects/RabbitMQ.Client/client/impl/Connection.cs

```diff
@@ -102,14 +102,9 @@ public Connection(IConnectionFactory factory, bool insist, IFrameHandler frameHa
             _factory = factory;
             _frameHandler = frameHandler;
 
-            if (factory is IAsyncConnectionFactory asyncConnectionFactory && asyncConnectionFactory.DispatchConsumersAsync)
-            {
-                ConsumerWorkService = new AsyncConsumerWorkService(factory.ConsumerDispatchConcurrency);
-            }
-            else
-            {
-                ConsumerWorkService = new ConsumerWorkService(factory.ConsumerDispatchConcurrency);
-            }
+            ConsumerWorkService = factory.DispatchConsumersAsync
+                ? new AsyncConsumerWorkService(factory.ConsumerDispatchConcurrency)
```

Comment:
I'm also wondering if there is a case to be made to simply remove the non-async consumers from 7.0 and keep them async-only?

Reply:
I think that is one discussion topic I was hopping to have in the meeting :D, then write it down and announce it to the community

Reply:
We can consider it.